src/main/scala/FilterCaptcha.scala

```diff
@@ -0,0 +1,68 @@
+import com.sksamuel.scrimage._
+import com.sksamuel.scrimage.filter._
+import java.awt.image.BufferedImage
+import java.awt.{Graphics2D,Color,Font}
+
+class FilterCaptcha extends CaptchaProvider {
+  val tokenAnswer = scala.collection.mutable.Map[String, String]()
+  def getChallenge(): Challenge = {
+    val filterTypes = List(new FilterType1, new FilterType2)
+    val r = new scala.util.Random
+    val alphabet = "abcdefghijklmnopqrstuvwxyz"
+    val n = 8
+    val answer = Stream.continually(r.nextInt(alphabet.size)).map(alphabet).take(n).mkString
+    val token = scala.util.Random.nextInt(10000).toString
+    tokenAnswer += token -> answer
+    val canvas = new BufferedImage(225, 50, BufferedImage.TYPE_INT_RGB)
+    val g = canvas.createGraphics()
+    g.setColor(Color.WHITE)
+    g.fillRect(0, 0, canvas.getWidth, canvas.getHeight)
+    g.setColor(Color.BLACK)
+    g.setFont(new Font("Serif", Font.PLAIN, 30))
+    g.drawString(answer, 5, 30)
+    g.dispose()
+    var image = new Image(canvas, ImageMetadata.empty)
+    val s = scala.util.Random.nextInt(2)
+    image = filterTypes(s).applyFilter(image)
+    val challenge = new Challenge(token, image)
+    challenge
+  }
+  def checkAnswer(token: String, input: String): Boolean = {
+    if(tokenAnswer(token)==input)
+    {
+      true
+    }
+    else
+    {
+      false
+    }
```

Comment:
Minor: too verbose.. you can reduce this `if else` block to the boolean expression: `tokenAnswer(token) == input`